libraries/FunctionalInterrupt/examples/Functional/Functional.ino

```diff
@@ -0,0 +1,88 @@
+#include <FunctionalInterrupt.h>
+
+#if defined(ESP32)
+#define BUTTON1 16
+#define BUTTON2 17
+#elif defined(ARDUINO_ESP8266_WEMOS_D1MINI)
+#define BUTTON1 D4
+#define BUTTON2 D3
+#else
+#define BUTTON1 2
+#define BUTTON2 0
+#endif
+
+class Button {
+  public:
+    Button(uint8_t reqPin) : PIN(reqPin) {
+      pinMode(PIN, INPUT_PULLUP);
+      // Arduino API:
+      //attachInterruptArg(PIN, [](void* self) {
+      //  static_cast<Button*>(self)->buttonIsr();
+      //}, this, FALLING); // fails on ESP8266: "buttonIsr not in IRAM"
+      //attachInterruptArg(PIN, reinterpret_cast<void(*)(void*)>(&buttonIsr_static), this, FALLING); // works on ESP8266
+      attachInterrupt(PIN, std::bind(&Button::buttonIsr, this), FALLING); // works on ESP8266
+      // FunctionalInterrupt.h API:
+      //attachScheduledInterrupt(PIN, [this](const InterruptInfo & ii) {
+      //  Serial.print("Pin ");
+      //  Serial.println(ii.pin);
+      //  buttonIsr();
+      //}, FALLING); // works on ESP8266
+    };
+    ~Button() {
+      detachInterrupt(PIN);
+    }
+
+#if defined(ESP8266)
+    void ICACHE_RAM_ATTR buttonIsr()
+#elif defined(ESP32)
+    void IRAM_ATTR buttonIsr()
```

Comment:
maybe, on top of the file, we can use
```
#ifndef IRAM_ATTR
#define IRAM_ATTR ICACHE_RAM_ATTR
#endif
```
that, because:
- this is an example in esp8266 repository
- `IRAM_ATTR` is now defined in esp8266 core master